File: mario.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{
    int height;

    // 1. Get a correct number from 1 to 8
    do
    {
        height = get_int("Height: ");
    }
    while (height < 1 || height > 8);

    // 2. Draw right-oriented piramid
    for (int i = 0; i < height; i++)
    {
        // print spaces from the left
        for (int j = 0; j < height - 1 - i; j++)
        {
            printf(" ");
        }

        // draw hashes
        for (int j = 0; j <= i; j++)
        {
            printf("#");
        }

        // go to the next row
        printf("\n");
    }
}

File: python/mario.py
def get_height() -> int:
    """Prompt the user for a height between 1 and 8 (inclusive)."""
    while True:
        try:
            height = int(input("Height: "))
            if 1 <= height <= 8:
                return height
        except ValueError:
            # Ignore non-numeric input and re-prompt
            pass

def draw_pyramid(height: int) -> None:
    """Draw a right-aligned pyramid of the given height."""
    for row in range(height):
        spaces = " " * (height - 1 - row)     # Leading spaces
        hashes = "#" * (row + 1)              # Pyramid blocks
        print(spaces + hashes)

def main() -> None:
    height = get_height()
    draw_pyramid(height)

if __name__ == "__main__":
    main()
